### ISSUE TITLE
quick guessing causes the hearts to not decresss

### ISSUE THREAD

**Issue by SAOfan** [l2lam/Word-guessing-game-demo#16]
not sure if grade 10+ but in grade 1 if you spam keys the heart assets stay and don't disappear after the first one until the the box saying it was wrong disappears although the game dose see them just the UI wont change might be down even when the box changes to show new letters the hearts won't go down until you stop guessing long enough for the box to disapear

**Comment by l2lam** [comment]
Good catch.  The deduction of lives is happening too late.  We should deduct lives (and do other state changes) immediately upon the detection of the incorrect guess; rather than wait for the rendering to happen to do the state changes.  The keyboard interrupts are asynchonous to the rendering so that is why this bug can happen.
